### DISCUSSION TITLE
Is  UWP platform abandoned?

### DISCUSSION THREAD

**Discussion by DJDQfff** [stride3d/stride#1514] [Q&A]
The latest version 4.1 can't export to uwp . And in some issues the team member mentioned that UWP will not support .

I don't know the details ,the roadmap can't find something about it.

I hope to get a specific answer .

Thanks.

**Comment by manio143** [comment]
In Stride 4.1 we decided to move to .NET 6 and start embracing new features of the CoreCLR that are not available in the .NET Standard that UWP supports. Moreover, Microsoft seems not to be investing in UWP anymore and has a [tool](https://docs.microsoft.com/en-us/dotnet/core/porting/upgrade-assistant-uwp-framework) to migrate off of it. You can read some people commenting on it [here](https://github.com/microsoft/WindowsAppSDK/discussions/1055).

It would be really hard to continue supporting UWP, because a lot of the fixes to the Stride build system and core dependencies would need to be half reversed. You can still use Stride 4.0 to develop UWP apps and if there has been a feature added or a bug fixed in 4.1 you can ask if it can be backported - that has been the case for a few things - unless it requires a lot of work to do so.

**Reply by DJDQfff** [reply]
Thank you ,I get it.

**Reply by yoshiask** [reply]
I understand that it seems like Microsoft isn't investing in UWP anymore, but it's still the best way to support Xbox and Hololens by a long shot. It's actually possible to use .NET 6 with UWP (and still be able to publish to the MS Store), but it's not officially supported and at least for XAML apps, requires a fork of C#/WinRT. If Stride isn't using XAML or anything from `Windows.UI.Xaml`, this might be an "easy" thing to support in 4.1 and beyond.

**Reply by Eideren** [reply]
We're still open to have PRs regarding this I'm pretty sure, still Stride is community driven so if you want to see support for it in the engine you or someone else in the community will have to take care of it

**Reply by tscrypter** [reply]
@yoshiask would you be willing to elaborate on the approach you're referring to use .NET 6 with UWP, in particular I'm interested in the fork of C#/WinRT. I was going to start tinkering with a PR for this. Happy to collaborate on it.

**Reply by yoshiask** [reply]
@tscrypter C#/WinRT has been hard coded by Microsoft to exclude the `Windows.UI.Xaml` (WUX) namespace from anything it generates. The reason for this is because of conflicts with WinUI 3, which has mostly the same types but in `Microsoft.UI.Xaml` (MUX). Since this is a UWP app, we can't use WinUI 3 so we'd need a WUX projection.

After I commented yesterday, I learned that the C#/WinRT fork is only necessary to generate a projection for WUX. It can be placed into a CI or something to automatically create the projection and then publish to NuGet. That way others looking to use WUX in .NET 6 don't have to worry about using the right C#/WinRT.

And even further, if Stride doesn't use anything from WUX, we don't even need a projection for it and UWP .NET 6 works without any real hacks.

**Reply by yoshiask** [reply]
After doing some research, I submitted a feature request for this with more details: #1517
